code/modules/input/focus.dm

```diff
@@ -0,0 +1,5 @@
+/mob/proc/set_input_focus(datum/new_focus)
+	if(input_focus == new_focus)
+		return
+	input_focus = new_focus
+	reset_perspective(input_focus) // Maybe this should be done manually? You figure it out, reader
```

Comment:
I'd say so yes. Changing the controlled character does not always equal changing the perspective